sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -354,9 +360,21 @@ private Mono<RxDocumentServiceResponse> toDocumentServiceResponse(Mono<HttpRespo
                     StoreResponse rsp = new StoreResponse(httpResponseStatus,
                         HttpUtils.unescape(httpResponseHeaders.toMap()),
                         content);
+
                     if (reactorNettyRequestRecord != null) {
                         rsp.setRequestTimeline(reactorNettyRequestRecord.takeTimelineSnapshot());
+
+                        rsp.setFaultInjectionRuleId(
+                            request
+                                .faultInjectionRequestContext
+                                .getFaultInjectionRuleId(reactorNettyRequestRecord.getTransportRequestId()));
+
+                        rsp.setFaultInjectionRuleEvaluationResults(
```

Comment:
So do we always set fault injection results on the response? what if fault injection is not enabled?

Reply:
it can be optimized to only do when fault injection is enabled. Will update in next PR

Reply:
updated